### PR TITLE
Upgrade to Docusaurus v3

### PR DESCRIPTION
**Pull request by ALRubinger**
Defined by #1274

### CODE REVIEW

**Comment by ALRubinger** [comment]
@dayhaysoos I think I've actually managed to get Docusaurus upgraded to the latest `3.1.1` release, along with its dependencies and a migration from MDX v1 to v3. Would you please do a PR review on this and see if it's good to merge?

This does a few things for us:
* Removes sec vuln transitive dependencies from our build
* Keeps us close to upstream for future upgrades
* Allows us (I think) to fix the remaining sec vulns because our existing Docusaurus version didn't play nicely with the forced upgrades we needed to put in place to correct the sec vulns. Or at least resolves this mutually exclusive problem for most of them.

File: site/docs/glossary.md [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 An agent is software that acts on behalf of a user to manage identity, public or private data, and interactions with other apps in a decentralized network. Agents hold a user's [DIDs](#decentralized-identifier-did), private keys, and a [DWN](#decentralized-web-node-dwn), and are permissioned to use the private keys of DIDs to act on a user's behalf to sign and store messages within a DWN.
 
-[<< More on Agents >>](/docs/web5/learn/agents)
+[\<\< More on Agents \>\>](/docs/web5/learn/agents)
```
Comment (angiejones) [inline site/docs/glossary.md:7]: what's the purpose of all the scape escapes?

File: site/docs/tbdex/issuer/vc-issuance.mdx [hunk 2]
```diff
@@ -24,20 +24,20 @@ If you need to create a Credential Issuance app from scratch, continue on with t
...
     </p>
   </div>
 </LanguageSwitchBlock>
 
-- Design the credential you'd like to offer.
-- Implement an issuance path where you'll accept a request for issuance, perform any necessary checks, and issue the credential.
-- Return the credential to the requester.
+\- Design the credential you'd like to offer.
+\- Implement an issuance path where you'll accept a request for issuance, perform any necessary checks, and issue the credential.
+\- Return the credential to the requester.
```
Comment (angiejones) [inline site/docs/tbdex/issuer/vc-issuance.mdx:40]: is markdown for bullets not outright supported in this version? 🤔

File: site/docs/tbdex/pfi/creating-offerings.mdx [hunk 3]
```diff
@@ -118,13 +106,13 @@ Note that while `type` and `description` fields are more intuitive and standard
...
+You can obtain this information via [Verifiable Credentials](/docs/web5/learn/verifiable-credentials).
 
-To specify the exact credentials you need, you can do so in the `requiredClaims` section of your Offering by following [creating a Presentation Definition](/docs/web5/build/verifiable-credentials/presentation-definition/). 
+To specify the exact credentials you need, you can do so in the `requiredClaims` section of your Offering by following [creating a Presentation Definition](/docs/web5/build/verifiable-credentials/presentation-definition/).
 You'll specify all required forms and details of proof, optional selection rules, and define flexibility where possible when different types of input may satisfy the requirements.
 
-For a full breakdown of all the available fields and combinations, [see the DIF website](https://identity.foundation/presentation-exchange/#presentation-definition) for details. 
+For a full breakdown of all the available fields and combinations, [see the DIF website](https://identity.foundation/presentation-exchange/#presentation-definition) for details.
```
Comment (angiejones) [inline site/docs/tbdex/pfi/creating-offerings.mdx:115]: is this just your linter updating all these files?

File: site/docs/tbdex/issuer/vc-issuance.mdx (hunk 2 above)
Comment (angiejones) [reply site/docs/tbdex/issuer/vc-issuance.mdx:40]: the bullets are all in one paragraph now

image

vs before

image

File: site/docs/glossary.md (hunk 1 above)
Comment (ALRubinger) [reply site/docs/glossary.md:7]: MDX v3, part of the Docusaurus 3 upgrade, is more restrictive than MDX v1:

https://docusaurus.io/docs/migration/v3#bad-usage-of--1

File: site/docs/tbdex/issuer/vc-issuance.mdx (hunk 2 above)
Comment (ALRubinger) [reply site/docs/tbdex/issuer/vc-issuance.mdx:40]: Great catch and thanks for the review @angiejones -- I'll add a commit to address all these.

File: site/docs/tbdex/pfi/creating-offerings.mdx (hunk 3 above)
Comment (ALRubinger) [reply site/docs/tbdex/pfi/creating-offerings.mdx:115]: Yep.

File: site/docs/tbdex/issuer/vc-issuance.mdx (hunk 2 above)
Comment (ALRubinger) [reply site/docs/tbdex/issuer/vc-issuance.mdx]: @angiejones Latest commit 95c95b8 should render now as:

image

File: site/docs/tbdex/issuer/vc-issuance.mdx [hunk 4]
```diff
@@ -23,21 +23,15 @@ If you need to create a Credential Issuance app from scratch, continue on with t
 
 <LanguageSwitchBlock>
   <div language="Kotlin">
-    <p>
-- Create a web server using <a href="https://www.jetbrains.com/help/idea/ktor.html#run_ktor_app">Ktor</a> that will serve as an API endpoint.
-    </p>
+    \* Create a web server using <a href="https://www.jetbrains.com/help/idea/ktor.html#run_ktor_app">Ktor</a> that will serve as an API endpoint.
```
Comment (angiejones) [inline site/docs/tbdex/issuer/vc-issuance.mdx:26]:
```suggestion
    * Create a web server using <a href="https://www.jetbrains.com/help/idea/ktor.html#run_ktor_app">Ktor</a> that will serve as an API endpoint.
```

Looks like we don't need this for bullets

**Review by angiejones — APPROVED** [review]

**Pull request merged**